acados/ocp_nlp/ocp_nlp_sqp.c

```diff
@@ -1229,10 +1229,9 @@ static void sqp_update_variables(void *config_, ocp_nlp_dims *dims, ocp_nlp_out
         blasfeo_daxpy(2*ni[i], alpha, mem->qp_out->t+i, 0, nlp_out->t+i, 0, nlp_out->t+i, 0);
 
         // linear update of algebraic variables using state and input sensitivity
-
         if (i < N)
         {
-            blasfeo_dgemv_t(nu[i]+nx[i], nz[i], 1.0, mem->dzduxt+i, 0, 0, nlp_out->ux+i, 0, 1.0, mem->z_alg+i, 0, nlp_out->z+i, 0); 
+            blasfeo_dgemv_t(nu[i]+nx[i], nz[i], 1.0, mem->dzduxt+i, 0, 0, nlp_out->ux+i, 0, 0.0, mem->z_alg+i, 0, nlp_out->z+i, 0); 
```

Comment:
Is this correct? it should be z^+ <- z + dzdx * delta_x + dzdu * delta_u

Reply:
where does the minus sign come from?

Reply:
oh sorry, it's an arrow, not a minus

Reply:
btw, these updates are both wrong, aren't they?

Reply:
Me and @zanellia had a chat about the correct fix, and I have an idea for a test to check if it is indeed correct. I will implement it in another PR soon.

Reply:
In my understanding `nlp_out->ux` does not contain `delta_x, delta_u`, but `x, u` looking at this line:
https://github.com/acados/acados/blob/2f3f19232ab14a69eb55179d5ed666db224b6b11/acados/ocp_nlp/ocp_nlp_sqp.c#L1215
Now thinking about this again, the correct update should be:
``blasfeo_dgemv_t(nu[i]+nx[i], nz[i], alpha, mem->dzduxt+i, 0, 0, mem->qp_out->ux+i, 0, 1.0, mem->z_alg+i, 0, nlp_out->z+i, 0); 
``
Does this make sense to you?

Reply:
Yes that's the correct one. Anyway, I quickly want to implement this fix and a check in another PR, and then you can merge it in your.